core/modules/file/file.pages.inc

```diff
@@ -1,9 +1,388 @@
+    $file_destination_dirname = backdrop_dirname($file_destination);
+    // Create the directory in case it doesn't exist.
+    file_prepare_directory($file_destination_dirname, FILE_CREATE_DIRECTORY);
+    if ($moved_file = file_move($file, $file_destination, FILE_EXISTS_RENAME)) {
+      // Only re-assign the file object if file_move() did not fail.
+      $file = $moved_file;
+    }
+  }
+
+  file_save($file);
+
+  $args = array(
+    '%title' => entity_label('file', $file),
+  );
+  watchdog('file', 'File: updated %title.', $args);
+  backdrop_set_message(t('File: %title has been updated.', $args));
+
+  // Clean up orphaned file.
+  if (!empty($orphaned_uri)) {
+    file_unmanaged_delete($orphaned_uri);
+
+    $args['@orphaned'] = file_uri_target($orphaned_uri);
+    watchdog('file', 'File: deleted orphaned file @orphaned for %title.', $args);
+    backdrop_set_message(t('The replaced @type @orphaned has been deleted.', $args));
+  }
+
+  $form_state['redirect'] = 'admin/content/files';
+
+  // Clear the page and block caches.
+  cache_clear_all();
+}
+
+/**
+ * Retrieves the upload validators for a file.
+ *
+ * @param array $options
+ *   (optional) An array of options for file validation.
+ *
+ * @return array
+ *   An array suitable for passing to file_save_upload() or for a managed_file
+ *   or upload element's '#upload_validators' property.
+ */
+function file_get_upload_validators(array $options = array()) {
+  // Set up file upload validators.
+  $validators = array();
+
+  // Validate file extensions. If there are no file extensions in $params there
+  // is no file extension validation.
+  if (!empty($options['file_extensions'])) {
+    $validators['file_validate_extensions'] = array($options['file_extensions']);
+  }
+  else {
+    $validators['file_validate_extensions'] = array(config_get('system.core', 'file_default_allowed_extensions'));
+  }
+
+  // Cap the upload size according to the system or user defined limit.
+  $max_filesize = parse_size(file_upload_max_size());
+  if (!empty($options['max_filesize']) && $options['max_filesize'] < $max_filesize) {
+    $max_filesize = parse_size($options['max_filesize']);
+  }
+
+  // There is always a file size limit due to the PHP server limit.
+  $validators['file_validate_size'] = array($max_filesize);
+
+  // Add image validators.
+  $options += array('min_resolution' => 0, 'max_resolution' => 0);
+  if ($options['min_resolution'] || $options['max_resolution']) {
+    $validators['file_validate_image_resolution'] = array($options['max_resolution'], $options['min_resolution']);
+  }
+
+  // Add other custom upload validators from options.
+  if (!empty($options['upload_validators'])) {
+    $validators += $options['upload_validators'];
+  }
+
+  return $validators;
+}
+
+/**
+ * Pre-render callback for adding validation descriptions to file upload fields.
+ */
+function file_upload_validators_pre_render($element) {
```

Comment:
Lets move this function into file.module, as it has generic usefulness.

Reply:
done.